docs/wiki/research/scaling/eip-4844.md

```diff
@@ -0,0 +1,220 @@
+
+See go-ethereum file "consensus.go" for checks added to the block header:
+
+- [link](https://github.com/ethereum/go-ethereum/blob/0a2f33946b95989e8ce36e72a88138adceab6a23/consensus/beacon/consensus.go#L274)
+
+### Changes on the Execution Specs
+
+The upgrade which will introduce EIP-4844 into the Execution Layer has been labeled **Cancun**.
+
+#### 1. Add checks inside the State Transition Function
+
+The EL now must check that the blob specific fields are valid for each transaction that is going to be executed in the State Transition Function (STF).
+
+The checks include (For the specs code, see [here](https://eips.ethereum.org/EIPS/eip-4844#execution-layer-validation)):
+
+- check that the signer has enough balance to cover the cost of both transaction gas fee and blob gas fees
+
+```python
+# modify the check for sufficient balance
+max_total_fee = tx.gas * tx.max_fee_per_gas
+if get_tx_type(tx) == BLOB_TX_TYPE:
+    max_total_fee += get_total_blob_gas(tx) * tx.max_fee_per_blob_gas
+assert signer(tx).balance >= max_total_fee
+```
+
+- check that the blob transaction contains at least 1 valid `blob_versioned_hash` (see CL changes) and that they are formatted correctly
+
+```python
+assert len(tx.blob_versioned_hashes) > 0
+```
+
+- check that the user is willing to pay at least the current blob base fee
+
+```python
+assert tx.max_fee_per_blob_gas >= get_blob_base_fee(block.header)
+```
+
+Finally, the EL STF must keep track of the gas being gas used for blob transactions (same as it already happens for EIP1559 transactions).
+
+```python
+blob_gas_used += get_total_blob_gas(tx)
+```
+
+## A note about previous attempts/research to improve scalability for rollups
+
+To accomplish scalability, the idea was initially to decrease transaction `calldata` gas cost with [EIP-4488](https://eips.ethereum.org/EIPS/eip-4488). However, this EIP is now irrelevant because EIP-4844 has been implemented instead. Here is a brief introduction for the EIP-4488.
+
+### EIP-4488
+
+Initally [EIP-4488](https://eips.ethereum.org/EIPS/eip-4488) was proposed to decrease the gas cost of `calldata`, and to add a limit of how much total transaction `calldata` can be in a block. The goal was to make Rollups transactions cheaper.
+Currently, the cost for a non zero byte of `calldata` costs 16 units of gas. With this EIP, the cost for a byte of `calldata` (regardless if is 0 or a non zero byte) would be 3 bytes.
+
+This would have introduced the following parameters:
+
+| Parameter                   | Value     |
+| --------------------------- | --------- |
+| NEW_CALLDATA_GAS_COST       | 3         |
+| BASE_MAX_CALLDATA_PER_BLOCK | 1,048,576 |
+| CALLDATA_PER_TX_STIPEND     | 300       |
+
+However, this EIP is now irrelevant because EIP-4844 has been implemented instead. Contrary to the original idea, the approach will be the exact opposite with [EIP-7623](https://eips.ethereum.org/EIPS/eip-7623), to increase `calldata` cost to decrease the maximum block size.
```

Comment:
add section for ## Further reading at the bottom. You can summarize more links, research posts, etc, lots of interesting stuff is in 4844 website https://www.eip4844.com/

Reply:
here is another great resource https://domothy.com/blobspace/

Reply:
added in the following [commit](https://github.com/eth-protocol-fellows/protocol-studies/pull/51/commits/ad9422b9a3aa6bb510f0fd76f2719bba0358bfdb)